ui/hooks/src/use-login.ts

```diff
@@ -187,18 +197,60 @@ export function useSignUp(provider: EthProviders, checkRegistered = false) {
   const connectWallet = useMutation(
     async () => lastValueFrom(sdk.api.auth.connectAddress(provider)),
     {
-      onError,
+      onError: (err: WalletTransactionError) => {
+        if (err.code === PROVIDER_ERROR_CODES.UserRejected) {
+          analyticsActions.trackEvent({
+            category: AnalyticsCategories.SIGN_UP_WALLET,
+            action: 'Step 1 Declined',
+          });
+        }
+        onError(err);
+      },
+      onSuccess: () => {
+        analyticsActions.trackEvent({
```

Comment:
I see that `analyticsActions` is marked as optional. I think you should add a guard here to check that `analyticsActions` is defined.